nixos/modules/services/networking/xrdp.nix

```diff
@@ -58,6 +62,25 @@ in
         package = mkPackageOptionMD pkgs "pulseaudio-module-xrdp" {};
       };
 
+      transportMode = mkOption {
+        type = types.enum [ null "unix" "tcp" "tcp6" "vsock" ];
+        default = null;
+        example = "tcp6";
+        description = lib.mdDoc ''
+          Specifies the transport mode on which the xrdp daemon listens. Null means all interfaces.
+        '';
+      };
+
+      address = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "192.168.1.1";
+        description = lib.mdDoc ''
+          Specifies the addresses on which the xrdp daemon listens. Null means all addresses.
+          Will be cid for vsock mode.
+        '';
+      };
+
```

Comment:
Might be out of scope for this PR, but xrdp supports [multiple listeners](https://github.com/neutrinolabs/xrdp/pull/1366) so this could probably be turned into a list of `{ transportMode, address }` attrsets.